searx/searx
# [mod] results.py: code clean up

## What does this PR do?

Simplify the code in results.py

---
About:
https://github.com/asciimoo/searx/blob/093dd42bb05feafb2cc77304d3b1a9a1a3016949/searx/results.py#L178-L183

this code:
* make sure that result['url'] is either None or an instance basestring (or subclass)
* then try to decode as if result['url'] is bytes instance.

It will always silently fail

---
About:
* https://github.com/asciimoo/searx/blob/093dd42bb05feafb2cc77304d3b1a9a1a3016949/searx/results.py#L178-L179
* https://github.com/asciimoo/searx/blob/093dd42bb05feafb2cc77304d3b1a9a1a3016949/searx/results.py#L184-L187

Isn't an error in the engine ?

----

This line:
https://github.com/asciimoo/searx/blob/093dd42bb05feafb2cc77304d3b1a9a1a3016949/searx/results.py#L172

```self.results``` is never used.

## Why is this change important?

More readable code.

## How to test this PR locally?

* Check if the results from different engines are merged
* Check if the results from an offline are parsed without error

## Author's checklist

N/A

## Related issues

No related issue
